src/components/VideoPlayerContexts/FullScreenContext.tsx

```diff
@@ -0,0 +1,34 @@
+import React, {useCallback, useContext, useMemo, useRef} from 'react';
+import type WindowDimensions from '@hooks/useWindowDimensions/types';
+import type ChildrenProps from '@src/types/utils/ChildrenProps';
+import type {FullScreenContext} from './types';
+
+const Context = React.createContext<FullScreenContext | null>(null);
+
+function FullScreenContextProvider({children}: ChildrenProps) {
+    const isFullscreen = useRef(false);
+    const lockedWindowDimensions = useRef<WindowDimensions | null>(null);
+
+    const lockWindowDimensions = useCallback((newWindowDimensions: WindowDimensions) => {
+        lockedWindowDimensions.current = newWindowDimensions;
+    }, []);
+
+    const unlockWindowDimensions = useCallback(() => {
+        lockedWindowDimensions.current = null;
+    }, []);
+
+    const contextValue = useMemo(() => ({isFullscreen, lockedWindowDimensions, lockWindowDimensions, unlockWindowDimensions}), [unlockWindowDimensions, lockWindowDimensions]);
```

Comment:
Im not 100% sure if I make sense, but here's a question:
if both `unlockWindowDimensions` and `lockWindowDimensions` have empty list `[]` in their respective `useCallback` calls - can they actually ever change?

Because if their references will never change, then putting them into this the dependency for `useMemo` does nothing?

Reply:
They won't ever change so probably leaving this useMemo list empty won't break anything. However It is advisable to specify inside `useMemo` for what function references changes it should look for. Also, without it ESlint is giving warning

Reply:
ok fair enough, if eslint prefers this then leave as is